src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -165,6 +172,131 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
+     * Information about a resource.
+     * - How many are needed to finish the build
+     * - How many are available to the builder
+     * - How many are in the player's inventory (client side only)
+     */
+    public static class BuildingBuilderResource
+    {
+        /**
+         * Availability status of the resource.
+         * according to the builder's chest, inventory and the player's inventory
+         */
+        public enum RessourceAvailability
+        {
+            NOT_NEEDED,
+            DONT_HAVE,
+            NEED_MORE,
+            HAVE_ENOUGH
+        }
+
+        private int amountAvailable;
+        private int amountPlayer;
+        private final String name;
+        private final ItemStack itemStack;
+        private final int amountNeeded;
+
+        public BuildingBuilderResource(final String name, final ItemStack itemStack, final int amountAvailable, final int amountNeeded)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document this public constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)